core/src/main/java/org/apache/carbondata/core/datamap/dev/DataMap.java

```diff
@@ -42,6 +42,17 @@
    */
   List<Blocklet> prune(FilterResolverIntf filterExp);
 
+  // TODO Move this method to Abstract class
+  /**
+   * Prune the datamap with filter expression and partition information. It returns the list of
+   * blocklets where these filters can exist.
+   *
+   * @param filterExp
+   * @return
+   */
+  List<Blocklet> prune(FilterResolverIntf filterExp, List<String> partitions);
+
+  // TODO Move this method to Abstract class
```

Comment:
This TODO is not valid now, right?

Reply:
In FG implementation I already created Abstract classes , so when we merge it we can move there